src/internal_telemetry/allocations/allocator/stack.rs

```diff
@@ -17,7 +17,7 @@ impl<const N: usize> GroupStack<N> {
     /// Creates an empty [`GroupStack`].
     pub const fn new() -> Self {
         Self {
-            current_top: 0,
+            current_top: 1,
```

Comment:
Can we just make this start at zero?

It should boil down to reordering the lines in `push` to increment before checking for overflow, and then the other methods can get a little simpler as well.